dist/src/main/java/io/camunda/application/commons/service/ManagementServicesConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.service;
+
+import io.camunda.application.commons.service.ManagementServicesConfiguration.ClientModeProperty;
+import io.camunda.application.commons.service.ManagementServicesConfiguration.LicenseKeyProperties;
+import io.camunda.service.ManagementServices;
+import io.camunda.service.license.CamundaLicense;
+import io.camunda.service.license.SaaSLicense;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties({LicenseKeyProperties.class, ClientModeProperty.class})
+public class ManagementServicesConfiguration {
+  private static final String SELF_MANAGED_MODE = "self-managed";
+  private final LicenseKeyProperties licenseKeyProperties;
+  private final ClientModeProperty clientModeProperty;
+
+  @Autowired
+  public ManagementServicesConfiguration(
+      final LicenseKeyProperties licenseKeyProperties,
+      final ClientModeProperty clientModeProperty) {
+    this.licenseKeyProperties = licenseKeyProperties;
+    this.clientModeProperty = clientModeProperty;
+  }
+
+  @Bean
+  public ManagementServices managementService(final CamundaLicense camundaLicense) {
+    final ManagementServices managementServices =
+        new ManagementServices().withLicense(camundaLicense);
+    // trigger to log if license is valid or not during startup
+    managementServices.isCamundaLicenseValid();
+
+    return managementServices;
+  }
+
+  @Bean
+  public CamundaLicense camundaLicense() {
+    final String clientMode = clientModeProperty.key();
+    if (SELF_MANAGED_MODE.equalsIgnoreCase(clientMode)) {
```

Comment:
I'll check with the PM, but I'll go with that approach for now

Reply:
As mentioned in the Slack message, if we tie the license check to this property, all you have to do in SM to avoid a license check is set the property to `saas` 🔐